src/snmalloc/pal/pal_linux.h

```diff
@@ -108,7 +113,15 @@ namespace snmalloc
         memset(p, 0x5a, size);
 
       madvise(p, size, MADV_DONTDUMP);
-      madvise(p, size, MADV_FREE);
+      madvise(
+        p,
+        size,
+#  if defined(MADV_FREE)
+        MADV_FREE
+#  else
+        MADV_DONTNEED
+#  endif
```

Comment:
I might have a very slight stylistic preference for this being made a `madvise_free_flags` `constexpr` or something rather than C preprocessor pragmas "inside" a function call.